### ISSUE TITLE
Snapchat

### ISSUE THREAD

**Issue by j0st** [wDCat/ANRC#7]
Does it work with Snapchat? Can you login even with xposed installed?

**Comment by darwinzp** [comment]
You have to try for yourself.

**Comment by wDCat** [comment]
I login normally and not saw any warning of xposed installed.

**Comment by fabbe50** [comment]
So what settings are you using and what version of Snapchat, been at this for hours trying to log in... they've also updated Snapchat yesterday, but I figured so I tried the previous versions (I.e 9.41.3.0, 9.41.2.0, 9.41.1.0), none of these versions I was able to log in. Always gives me the same error.

**Comment by wDCat** [comment]
@fabbe50  can you update a screenshot of this error msg?Thx~

**Comment by fabbe50** [comment]
![screenshot_20161022-143114](https://cloud.githubusercontent.com/assets/8047187/19619306/57982bd2-9864-11e6-9501-6afca5e55f40.png)

Here you go! :)

**Comment by dinaps86** [comment]
I think they must have updated snapchat to not even let you login if device is rooted (*I think it only looked for xposed in past versions, but now it won't ebe let me login though my htc one m9 which is rooted but not running xposed) there's gotta be some sort of fix for this

**Comment by fabbe50** [comment]
According to some forum posts (old posts) they are checking if they can find root directories or certain root apps (SuperSU, Xposed Installer etc.). My biggest issue is to figure out what directories and what apps it's checking for. Seeing as I can't log in to any of my rooted devices (with or without Xposed), I think your conclusion is correct.  I think there's a solution to everything, just that I feel like solving this could be hard.

**Comment by wDCat** [comment]
yeah .i decompile Snapchat and only found some check code in java.

see http://pastebin.com/AsHxJmcK

for bypassing root detecting in java,try this app:

http://repo.xposed.info/module/com.devadvance.rootcloak2

**Comment by fabbe50** [comment]
Doesn't seem to be working, I don't know if there's something more to it. Do you have the possibility to send me the part of the code which executes when you login, so I could take a look. Thanks!

**Comment by fabbe50** [comment]
Update: I've tried cloaking root with the latest version of rootcloak, but I do think there's more to it. Since it's failing during the stage where Snapchat says logging in and not where it says verifying device. So it aborts login after the text has been changed to logging in. Something else is happening during the login stage than what's happening during the verifying device stage.

**Comment by NikoGrub** [comment]
For me it Snapchat isn't working with RootCloak and ANRC. With ARNC enabled, the log in process stuck at "verifying device".  I hope someone will find a solution for that.

**Comment by NikoGrub** [comment]
Sorry for the autocorrection..

**Comment by fabbe50** [comment]
I'm currently trying to reverse engineer the app, hiding the different paths and commands that Snapchat is checking for doesn't seem to work. Please tell me if you find or have anything of use, could be a logcat or a reference to a class etc.. I will make this work!

**Comment by NikoGrub** [comment]
Sadly I don't know how to create a logcat :( I downloaded aLogcat, but I don't know how to create a logcat just for Snapchat.

**Comment by wDCat** [comment]
Does Snapchat use Safetynet too?_(:зゝ∠)_

**Comment by fabbe50** [comment]
Slowly starting to feel like it. :/

**Comment by fabbe50** [comment]
After 2 days of testing I can confirm Snapchat using SafetyNet. Also took a look into how the SafetyNet API works and came to the conclusion that it can be fooled, I took a look online and found that it has been fooled on multiple occasions by two pieces of software named SUHide and Magisk. So I would suggest taking a look at those two pieces of software, because plain old root hiding wont suffice against SafetyNet checks.
